plugins/by-name/alpha/default.nix

```diff
@@ -156,20 +165,24 @@ in
       opt = options.plugins.alpha;
     in
     lib.mkIf cfg.enable {
-      # TODO: deprecated 2024-08-29 remove after 24.11
-      warnings = lib.mkIf opt.iconsEnabled.isDefined [
```

Comment:
By dropping this we no longer warn when someone defines `iconsEnabled = false`.

If the option truly is deprecated, then any definition should have a warning. It's fine to have different warnings based on additional things though, and/or to build the warning text dynamically based on additional factors.

Reply:
Doh, you're right...

Reply:
I was trying to consolidate the message down so there weren't so many warnings for what I thought was that same thing.

Reply:
Yeah, ideally we'd only print one warning message. I don't mind if it's one "dumb" all encompassing messages, or if it takes additional conditions into account, but we should definitely always have _some_ warning when the option is defined.